packages/playout-gateway/src/tsrHandlerJobs/initDeviceJob.ts

```diff
@@ -0,0 +1,201 @@
+				// Todo: send these to Core
+				this.logger.info('commandReport', {
+					commandReport: commandReport,
+				})
+			}
+		}
+		const onCommandError = (error: any, context: any) => {
+			// todo: handle this better
+			this.logger.error(error)
+			this.logger.debug(context)
+		}
+		const onUpdateMediaObject = (collectionId: string, docId: string, doc: MediaObject | null) => {
+			coreTsrHandler.onUpdateMediaObject(collectionId, docId, doc)
+		}
+		const onClearMediaObjectCollection = (collectionId: string) => {
+			coreTsrHandler.onClearMediaObjectCollection(collectionId)
+		}
+		const fixError = (e: any): string => {
+			const name = `Device "${deviceContainer.deviceName || this.deviceId}" (${deviceContainer.instanceId})`
+			if (e.reason) e.reason = name + ': ' + e.reason
+			if (e.message) e.message = name + ': ' + e.message
+			if (e.stack) {
+				e.stack += '\nAt device' + name
+			}
+			if (_.isString(e)) e = name + ': ' + e
+
+			return e
+		}
+
+		deviceContainer.onChildClose = () => {
+			// Called if a child is closed / crashed
+			this.logger.warn(`Child of device ${this.deviceId} closed/crashed`)
+			debug(`Trigger update devices because "${this.deviceId}" process closed`)
+
+			onDeviceStatusChanged({
+				statusCode: StatusCode.BAD,
+				messages: ['Child process closed'],
+			})
+
+			this.tsrHandler.onDeviceChildClosed(this.deviceId)
+		}
+		// Note for the future:
+		// It is important that the callbacks returns void,
+		// otherwise there might be problems with threadedclass!
+
+		await deviceContainer.device.on('connectionChanged', onDeviceStatusChanged as () => void)
+		// await device.device.on('slowCommand', onSlowCommand)
+		await deviceContainer.device.on('slowSentCommand', onSlowSentCommand as () => void)
+		await deviceContainer.device.on('slowFulfilledCommand', onSlowFulfilledCommand as () => void)
+		await deviceContainer.device.on('commandError', onCommandError as () => void)
+		await deviceContainer.device.on('commandReport', onCommandReport as () => void)
+		await deviceContainer.device.on('updateMediaObject', onUpdateMediaObject as () => void)
+		await deviceContainer.device.on('clearMediaObjects', onClearMediaObjectCollection as () => void)
+
+		await deviceContainer.device.on('info', ((e: any, ...args: any[]) => {
+			this.logger.info(fixError(e), ...args)
+		}) as () => void)
+		await deviceContainer.device.on('warning', ((e: any, ...args: any[]) => {
+			this.logger.warn(fixError(e), ...args)
+		}) as () => void)
+		await deviceContainer.device.on('error', ((e: any, ...args: any[]) => {
+			this.logger.error(fixError(e), ...args)
+		}) as () => void)
+
+		await deviceContainer.device.on('debug', (...args: any[]) => {
+			if (!deviceContainer.debugLogging && !this.tsrHandler.coreHandler.logDebug) {
+				return
+			}
+			if (args.length === 0) {
+				this.logger.debug('>empty message<')
+				return
+			}
+			const data = args.map((arg) => (typeof arg === 'object' ? JSON.stringify(arg) : arg))
```

Comment:
Parantheses around `typeof arg === 'object' ? JSON.stringify(arg) : arg` are unnecessary.

Reply:
But they are required by the linter 
![image](https://user-images.githubusercontent.com/9103996/210366518-6badb7c4-993f-454a-86e0-b4ab41114bb3.png)